src/OpenTelemetry.AutoInstrumentation.Native/member_resolver.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef OTEL_CLR_PROFILER_MEMBER_RESOLVER_H_
+#define OTEL_CLR_PROFILER_MEMBER_RESOLVER_H_
+
+#include <vector>
+#include <corhlpr.h>
+
+#include "com_ptr.h"
+#include "module_metadata.h"
+
+namespace trace
+{
+
+class MemberResolver
+{
+private:
+    ComPtr<IMetaDataImport2> metadaImport_;
+    ComPtr<IMetaDataEmit2>   metadaEmit_;
+
+public:
+    MemberResolver(const ComPtr<IMetaDataImport2>& import, const ComPtr<IMetaDataEmit2>& emit);
+
+
+    /// <summary>
+    /// Uses IMetaDataImport::FindTypeDefByName if tkResolutionScope is mdTokenNill, TypeDef or Assembly
+    /// Uses IMetaDataEmit::DefineTypeRefByName otherwise (if tkResolutionScope is AssemblyRef or TypeRef)
+    /// </summary>
+    HRESULT GetTypeRefOrDefByName(mdToken tkResolutionScope, LPCWSTR szName, mdToken* token);
+
+    /// <summary>
+    /// Uses IMetaDataImport::FindTypeDefByName if tkResolutionScope is mdTokenNill, TypeDef or Assembly
+    /// Uses IMetaDataEmit::DefineTypeRefByName otherwise (if tkResolutionScope is AssemblyRef or TypeRef)
```

Comment:
I'm not sure if my suggestion is 100% accurate, can you follow-up here?

```suggestion
    /// Uses IMetaDataImport::FindMember if tkResolutionScope is TypeDef
    /// Uses IMetaDataEmit::DefineMemberRef otherwise (if tkResolutionScope is mdTokenNill, Assembly, AssemblyRef, or TypeRef)
```